docs/source/grpo_trainer.md

```diff
@@ -524,6 +526,67 @@ and the reward will be computed as the sum of the rewards from each function, or
 
 Note that [`GRPOTrainer`] supports multiple reward functions of different types. See the parameters documentation for more details.
 
+## Vision Language Model (VLM) Training
+
+GRPO supports training Vision Language Models (VLMs) with comprehensive image processing capabilities. This feature enables training on multimodal datasets that include both text and images.
+
+### Key Features
+
+- **Automatic VLM Detection**: GRPO automatically detects VLM processors (those with both `tokenizer` and `image_processor` attributes) and configures appropriate data handling
+- **Image Processing**: Supports various image formats and automatically processes images through the model's image processor
+- **vLLM Integration**: Full compatibility with vLLM acceleration in both `server` and `colocate` modes for VLMs
+- **Memory Optimization**: Supports 4-bit quantization with LoRA for memory-efficient VLM training
+- **Enhanced Data Collation**: Proper handling of VLM-specific data structures including `pixel_values`, `pixel_attention_mask`, and `image_sizes`
+
+### Supported VLM Models
+
+The implementation has been tested with popular VLM architectures including:
+- **Qwen2.5-VL**: `Qwen/Qwen2.5-VL-3B-Instruct`
```

Comment:
Nice
I'll create tiny models, it will be easier for testing

Reply:
would anyone kindly share the setup for running test script with `Qwen2.5-VL-3B-Instruct` please? I tried the parameters from the example script docstring, getting cuda oom with 8 x A100 40GB which feels wrong

Reply:
For that model, minimal tested config (completed successful train run):
* Example script, 1 A100 80GB.
* [Notebook/recipe](https://github.com/huggingface/trl/pull/3072#issuecomment-3062854034), 1 A100 40GB (Colab Pro).